sympy/solvers/ode/tests/test_riccati.py

```diff
@@ -0,0 +1,903 @@
+        match, funcs = match_riccati(eq, f, x)
+        assert match == res
+        if res:
+            assert [b0, b1, b2] == funcs
+
+
+def test_poles():
+    tests = [
+    (
+        Poly(3*x**2 - x, x),
+        {0: 1, S(1)/3: 1}
+    ),
+    (
+        Poly((2 - x)**2*(2*x - 1)**2, x),
+        {2: 2, S(1)/2: 2}
+    ),
+    (
+        Poly(x**2 - x - 1, x),
+        {(1 + sqrt(5))/2: 1, (1 - sqrt(5))/2: 1}
+    ),
+    (
+        Poly(x + 2, x),
+        {-2: 1}
+    ),
+    (
+        Poly(x**4 + 3*x**3 - 3*x**2 + 3*x - 4, x),
+        {I: 1, -I: 1, 1: 1, -4: 1}
+    ),
+    (
+        Poly(x**5 - 4*x**4 + 4*x**3 + x**2 - 4*x + 4, x),
+        {2: 2, -1: 1, (1 - sqrt(3)*I)/2: 1, (1 + sqrt(3)*I)/2: 1}
+    ),
+    (
+        Poly(x**3 + 12*x**2 + 39*x + 28, x),
+        {-1: 1, -4: 1, -7: 1}
+    ),
+    (
+        Poly(x**4 + 6*x**3 + 11*x**2 + 6*x + 1, x),
+        {-S(3)/2 - sqrt(5)/2: 2, -S(3)/2 + sqrt(5)/2: 2}
+    ),
+    (
+        Poly(x**5 + 5*x**4 + 10*x**3 + 10*x**2 + 3*x + 1, x),
+        {}
+    ),
+    (
+        Poly(2*x**7 + 6*x**6 + 11*x**5 - 12*x**4 - 28*x**3 + 6*x**2 + 15*x, x),
+        {1: 2, -1: 2, -S(3)/2 - sqrt(21)*I/2: 1, -S(3)/2 + sqrt(21)*I/2: 1, 0: 1}
+    )]
+    for den, poles in tests:
+        assert find_poles(den, x) == poles
+
+
+def test_val_at_inf():
+    tests = [
+    (
+        Poly(12*x**8 - 12*x**7 - 11*x**6 + 8*x**5 + 3*x**4 - x**3 + x**2 - 11*x, x),
+        Poly(-14*x**2 + x, x),
+         -6
+    ),
+    (
+        Poly(-8*x**4 + 14*x**3 + 10*x**2 + 3*x - 11, x),
+        Poly(-9*x**4 + 3*x**3 + 15*x**2 - 6*x - 14, x),
+         0
+    ),
+    (
+        Poly(-6*x**3 - 8*x**2 + 8*x - 6, x),
+        Poly(-5*x**3 + 12*x**2 - 6*x - 9, x),
+         0
+    ),
+    (
+        Poly(15*x**3 - 5*x**2 - 6*x - 2, x),
+        Poly(9*x**5 + 12*x**4 + 7*x**3 + 11*x**2 - 2*x - 9, x),
+         2
+    ),
+    (
+        Poly(10*x**3 + 8*x**2 - 13*x + 6, x),
+        Poly(-13*x**10 - x**9 + 5*x**8 + 7*x**7 + 10*x**6 + 6*x**5 - 7*x**4 + 11*x**3 - 8*x**2 + 5*x + 13, x),
+         7
+    ),
+    (
+        Poly(-15*x - 15, x),
+        Poly(15*x**3 - 13*x**2 - 4*x + 2, x),
+         2
+    ),
+    (
+        Poly(-9*x**8 - 9*x**7 + 11*x**6 - 11*x**5 + 9*x**4 + 7*x**3 - 8*x**2 - 5*x - 12, x),
+        Poly(9*x**10 + 4*x**9 + 12*x**8 - 8*x**7 - 3*x**6 - 14*x**5 + 8*x**4 - 15*x**3 + 4*x**2 + 9*x + 1, x),
+         2
+    ),
+    (
+        Poly(13*x - 9, x),
+        Poly(-8*x**4 + 5*x**3 - 4*x**2 - 14*x + 4, x),
+         3
+    ),
+    (
+        Poly(12*x**7 - 7*x**6 + 9*x**5 + 6*x**4 - 14*x**3 + 15*x**2 - x + 8, x),
+        Poly(11*x**10 + 2*x**9 + x**8 - 3*x**7 - 12*x**6 - 10*x**5 - 15*x**4 - 14*x**3 - 5*x**2 + 15*x + 15, x),
+         3
+    ),
+    (
+        Poly(5*x**6 + 9*x**5 - 11*x**4 - 9*x**3 + x**2 - 4*x + 4, x),
+        Poly(15*x**4 + 3*x**3 - 8*x**2 + 15*x + 12, x),
+         -2
+    )]
+    for num, den, val in tests:
+        assert val_at_inf(num, den, x) == val
```

Comment:
Again I don't see why so many examples are needed. This function is basically trivial. It would be more useful to test base cases like what happens if the numerator is zero or do we even care about that case?

Reply:
I'll remove some examples. The numerator being zero would mean `a(x)` is 0, and the solution is trivial for this case. So, that wouldn't be of interest here.